osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -41,6 +41,9 @@ public BeatmapInfo BeatmapInfo
             get => BeatmapInfoBacking;
             set
             {
+                if (BeatmapInfoBacking.Equals(value))
+                    return;
```

Comment:
This can be potentially bypassed if `BeatmapInfo` was resolved from realm with different `BeatmapInfoBacking.Hash` and `BeatmapHash` and we set a beatmap which is equal to `BeatmapInfoBacking`.

Reply:
Okay, so don't do that?

As developers, we have something called "common sense", which along with good commenting and insight of the surrounding code, can easily avoid mistakes even if it's not guarded by the code.

This would be different if this was an API project, but `osu.Game` is the final point of interaction and people like myself are in control of review to ensure errors like the one you mention will (usually) not occur.